engine/src/bmr/BufferProvider.cpp

```diff
@@ -131,13 +131,15 @@ std::unique_ptr<blazing_allocation_chunk> allocation_pool::get_chunk() {
 
 
 void allocation_pool::grow() {
-
   // if this is the first growth (initializaton) then we want num_buffers, else we will just grow by half that.
   std::size_t num_new_buffers = this->buffer_counter == 0 ? this->num_buffers : this->num_buffers/2;
   allocations.push_back(std::make_unique<blazing_allocation>());
+  allocations.back()->index = this->allocations.size() - 1;
+  allocations.back()->total_number_of_chunks = num_new_buffers;
   auto last_index = allocations.size() -1;
   try{
     allocator->allocate((void **) &allocations[last_index]->data,num_new_buffers * buffer_size);
+    this->allocations[last_index]->total_number_of_chunks = num_new_buffers;
```

Comment:
You are setting `total_number_of_chunks` twice. Dont do that. Delete this one or the one above

Reply:
done